libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -195,65 +189,51 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     }
 
     private fun sessionStart() {
-        var freeDrive = false
+        var isActiveGuidance = true
         ifNonNull(weakMapboxNavigation.get()?.getRoutes()) { routes ->
-            freeDrive = routes.isEmpty()
+            isActiveGuidance = routes.isNotEmpty()
         }
-        when (freeDrive) {
+        when (isActiveGuidance) {
             true -> {
-                // Do nothing
-                Log.d(TAG, "Free drive not supported")
-            }
-            false -> {
                 telemetryThreadControl.scope.launch {
                     callbackDispatcher.resetRouteProgressProcessor()
                     postUserEventDelegate =
                         postUserFeedbackEventAfterInit // Telemetry is initialized and the user selected a route. Allow user feedback events to be posted
                     handleSessionStart()
-                    sessionEndPredicate = { sessionStop() }
+                    sessionEndPredicate = {
+                        telemetryScope.launch {
+                            sessionStop()
+                        }
+                    }
                 }
             }
+            false -> {
+                // Do nothing
+                Log.d(TAG, "Only Active Guidance supported")
+            }
         }
     }
 
-    private fun sessionStop() {
-        Log.d(TAG, "TripSessionState.STOPPED")
+    private suspend fun sessionStop() {
```

Comment:
Would we want to remove the `suspend` and handle the launch inside `sessionStop`?

Reply:
If you look at it carefully `sessionStop` is executed when `sessionEndPredicate` lambda is executed, so that'd require extra thought and changes. Didn't want to change the overall logic, only wanted to fix the outstanding issues 😅

Reply:
ah it's also for the `.join()`

Maybe this would be better? There's another job controller in here 
```
private fun sessionStop() = telemetryScope.launch {
```